### PR TITLE
Update @nextcloud/dialogs and fix toasts styling

### PR DESCRIPTION
**Pull request by skjnldsv**
https://github.com/nextcloud/nextcloud-vue/pull/3402#discussion_r1007702975

- [x] Needs https://github.com/nextcloud/nextcloud-dialogs/pull/694

### CODE REVIEW

File: core/css/server.css [hunk 1]
```diff
@@ -4887,56 +4890,67 @@ kbd {
   font-size: 0;
   /* dark theme overrides for Nextcloud 25 and later */
 }
-.toastify.dialogs .toast-undo-button.toast-close::before,
-.toastify.dialogs .toast-close.toast-close::before {
-  background-image: url("./close.svg");
+
+.toast_toastify__VG2Rg.toast_dialogs__rG5MI .toast_toast-undo-button__-zxpv.toast_toast-close__jbbSJ::before,
+.toast_toastify__VG2Rg.toast_dialogs__rG5MI .toast_toast-close__jbbSJ.toast_toast-close__jbbSJ::before {
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMTYiIHdpZHRoPSIxNiI+CiAgPHBhdGggZD0iTTE0IDEyLjNMMTIuMyAxNCA4IDkuNyAzLjcgMTQgMiAxMi4zIDYuMyA4IDIgMy43IDMuNyAyIDggNi4zIDEyLjMgMiAxNCAzLjcgOS43IDh6Ii8+Cjwvc3ZnPgo=");
```
Comment (skjnldsv) [inline core/css/server.css:4896]: Important part

File: core/css/server.css [hunk 2]
```diff
@@ -4887,56 +4890,67 @@ kbd {
...
 /* dark theme overrides for Nextcloud 24 and earlier */
-.theme--dark .toastify.dialogs .toast-close {
+.toast_theme--dark__nNWX9 .toast_toastify__VG2Rg.toast_dialogs__rG5MI .toast_toast-close__jbbSJ {
   /* close icon style */
 }
-.theme--dark .toastify.dialogs .toast-close.toast-close::before {
-  background-image: url("./close-dark.svg");
+
+.toast_theme--dark__nNWX9 .toast_toastify__VG2Rg.toast_dialogs__rG5MI .toast_toast-close__jbbSJ.toast_toast-close__jbbSJ::before {
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMTYiIHdpZHRoPSIxNiI+CiAgPHBhdGggZD0iTTE0IDEyLjNMMTIuMyAxNCA4IDkuNyAzLjcgMTQgMiAxMi4zIDYuMyA4IDIgMy43IDMuNyAyIDggNi4zIDEyLjMgMiAxNCAzLjcgOS43IDh6IiBzdHlsZT0iZmlsbC1vcGFjaXR5OjE7ZmlsbDojZmZmZmZmIi8+Cjwvc3ZnPgo=");
```
Comment (skjnldsv) [inline core/css/server.css:4953]: Important part 2

**Comment by skjnldsv** [comment]
Done, please review

File: core/css/server.css [hunk 3]
```diff
@@ -4887,46 +4890,56 @@ kbd {
   font-size: 0;
   /* dark theme overrides for Nextcloud 25 and later */
 }
+
 .toastify.dialogs .toast-undo-button.toast-close::before,
 .toastify.dialogs .toast-close.toast-close::before {
-  background-image: url("./close.svg");
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMTYiIHdpZHRoPSIxNiI+CiAgPHBhdGggZD0iTTE0IDEyLjNMMTIuMyAxNCA4IDkuNyAzLjcgMTQgMiAxMi4zIDYuMyA4IDIgMy43IDMuNyAyIDggNi4zIDEyLjMgMiAxNCAzLjcgOS43IDh6Ii8+Cjwvc3ZnPgo=");
```
Comment (skjnldsv) [inline core/css/server.css:4896]: Important part

File: core/css/server.css [hunk 4]
```diff
@@ -4935,10 +4948,12 @@ kbd {
 .theme--dark .toastify.dialogs .toast-close {
   /* close icon style */
 }
+
 .theme--dark .toastify.dialogs .toast-close.toast-close::before {
-  background-image: url("./close-dark.svg");
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMTYiIHdpZHRoPSIxNiI+CiAgPHBhdGggZD0iTTE0IDEyLjNMMTIuMyAxNCA4IDkuNyAzLjcgMTQgMiAxMi4zIDYuMyA4IDIgMy43IDMuNyAyIDggNi4zIDEyLjMgMiAxNCAzLjcgOS43IDh6IiBzdHlsZT0iZmlsbC1vcGFjaXR5OjE7ZmlsbDojZmZmZmZmIi8+Cjwvc3ZnPgo=");
```
Comment (skjnldsv) [inline core/css/server.css:4953]: Important part 2

File: core/css/server.scss [hunk 5]
```diff
@@ -7,5 +7,6 @@
 @import 'fixes.scss';
 @import 'mobile.scss';
 @import 'tooltip.scss';
-@import '../../node_modules/@nextcloud/dialogs/styles/toast.scss';
+// If you include .css, it will be imported as url
+@import '../../node_modules/@nextcloud/dialogs/dist/index';
```
Comment (artonge) [inline core/css/server.scss:11]: Maybe specifying loaders inline would prevent this 'hack'
https://webpack.js.org/concepts/loaders/#inline

**Review by artonge — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
🐘

**Comment by szaimen** [comment]
ESLint is failing...

File: core/css/server.scss (hunk 5 above)
Comment (skjnldsv) [reply core/css/server.scss:11]: this is not webpack

Comment (artonge) [reply core/css/server.scss:11]: :see_no_evil:

Comment (skjnldsv) [reply core/css/server.scss:11]: :grin: :hugs:

**Comment by skjnldsv** [comment]
> ESLint is failing...

https://github.com/nextcloud/nextcloud-dialogs/pull/699

**Comment by skjnldsv** [comment]
AAaaaaaaand it needs a rebase...

**Pull request merged**